### PR TITLE
Controller/load save criteria

### PR DESCRIPTION
**Pull request by MelissaRoman06**
Save criteria.
Delete criteria.
Load criteria implemented

### CODE REVIEW

File: src/main/java/com/fundation/logic/database/Query.java [hunk 1]
```diff
@@ -79,30 +83,36 @@ public void deleteByIde(String ID) {
...
-            Connection connection = DBConnection.getInstance().getConnection();
-            PreparedStatement statement = connection.prepareStatement(sql);
-            statement.setTimestamp(1, Timestamp.valueOf(lowerdt));
-            statement.setTimestamp(2, Timestamp.valueOf(upperdt));
-            ResultSet result = statement.executeQuery(sql);
-            while (result.next()) {
-                infCriterias.add(result.getInt("id") + "\t" + result.getString("name") + "\t" + result.getDate("date") + "\t" + result.getString("json"));
+            DateFormat dateFormatStart = new SimpleDateFormat( "EEE MMM dd HH:mm:ss zzz yyyy", Locale.US);
+            Date date1 = dateFormatStart.parse(firstDate);
+            System.out.println(date1);
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/database/Query.java]: delete this print

File: src/main/java/com/fundation/logic/database/Query.java [hunk 2]
```diff
@@ -79,30 +83,36 @@ public void deleteByIde(String ID) {
...
+            DateFormat dateFormatEnd = new SimpleDateFormat( "EEE MMM dd HH:mm:ss zzz yyyy", Locale.US);
+            Date date2 = dateFormatEnd.parse(secondDate);
+            System.out.println(date2);
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/database/Query.java]: delete this print

File: src/main/java/com/fundation/logic/model/search/CommonSearch.java [hunk 3]
```diff
@@ -101,7 +101,7 @@ public List searchInPath(String path) {
                             evaluateString(owner, criteriaOwner) &&
                             evaluateString(mimeType, criteriaMimeType)){
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/model/search/CommonSearch.java]: put space after ")"

File: src/main/java/com/fundation/logic/view/loadSaveCriteria/PopupLoadSave.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    }
+
+    /**
+     * Allows to get delete item from the menu.
+     * @return Delete item.
+     */
+    public JMenuItem getDeleteItem() {
+        return deleteItem;
+    }
+/**
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/view/loadSaveCriteria/PopupLoadSave.java]: this code are commented

File: src/test/java/com/fundation/logic/model/QueryForCriteriaTest.java [hunk 5]
```diff
@@ -44,12 +45,15 @@ public void saveCriteria_createNewCriteria_saveOnDB_compareToLastOne() {
...
-            "2019-07-16", "Common", "{\"path\":\"resources/\"}");
-        assertTrue(expectedCriteriaRecord.equals(actualCriteriaRecord));
-        queryForCriteria.deleteById(idLastCriteria + 1);
+        CriteriaRecord expectedCriteriaRecord = new CriteriaRecord(idLastString, "Criteria Test",
+            "2019-07-29", "Common", "{\"path\":\"resources/\"}");
+        assertEquals(expectedCriteriaRecord.getJson(), actualCriteriaRecord.getJson());
+        assertEquals(expectedCriteriaRecord.getId(), actualCriteriaRecord.getId());
+        queryForCriteria.deleteById(idLastString);
     }
 }
```
Comment (titor-oopart) [inline src/test/java/com/fundation/logic/model/QueryForCriteriaTest.java]: put a empty line

File: src/main/java/com/fundation/logic/view/loadSaveCriteria/PopupLoadSave.java (hunk 4 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/view/loadSaveCriteria/PopupLoadSave.java]: done

File: src/test/java/com/fundation/logic/model/QueryForCriteriaTest.java (hunk 5 above)
Comment (MelissaRoman06) [reply src/test/java/com/fundation/logic/model/QueryForCriteriaTest.java]: done

File: src/main/java/com/fundation/logic/model/search/CommonSearch.java (hunk 3 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/model/search/CommonSearch.java]: done

File: src/main/java/com/fundation/logic/database/Query.java [hunk 6]
```diff
@@ -79,30 +85,38 @@ public void deleteByIde(String ID) {
...
-            ResultSet result = statement.executeQuery(sql);
-            while (result.next()) {
-                infCriterias.add(result.getInt("id") + "\t" + result.getString("name") + "\t" + result.getDate("date") + "\t" + result.getString("json"));
+            DateFormat dateFormatStart = new SimpleDateFormat( "EEE MMM dd HH:mm:ss zzz yyyy", Locale.US);
+            Date date1 = dateFormatStart.parse(firstDate);
+            System.out.println(date1);
+            LocalDateTime startDate = date1.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
+            System.out.println(startDate);
+            DateFormat dateFormatEnd = new SimpleDateFormat( "EEE MMM dd HH:mm:ss zzz yyyy", Locale.US);
+            Date date2 = dateFormatEnd.parse(secondDate);
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/database/Query.java]: use a variable name more understandable for date2 variable, for example, auxDate or secondDate.

File: src/main/java/com/fundation/logic/database/Query.java [hunk 7]
```diff
@@ -79,30 +85,38 @@ public void deleteByIde(String ID) {
...
+            LocalDateTime endDate = date2.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime();
+            System.out.println(endDate);
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/database/Query.java]: delete SOUT

File: src/main/java/com/fundation/logic/main.java [hunk 8]
```diff
@@ -22,11 +23,15 @@
...
 
     public static void main(String[] arg) {
-        MainFrame searchFrame = new MainFrame();
-        Controller controller = new Controller(searchFrame);
-        ConvertController convertController = new ConvertController(searchFrame);
+        MainFrame mainFrame = new MainFrame();
+        SearchController searchController = new SearchController(mainFrame);
+        ConvertController convertController = new ConvertController(mainFrame);
+        LoadSaveController loadSaveController = new LoadSaveController(mainFrame, searchController);
+        searchController.listenSearchButtons();
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/main.java]: Try to insert these calls to constructors of classes.

File: src/main/java/com/fundation/logic/model/QueryForCriteria.java [hunk 9]
```diff
@@ -47,49 +48,93 @@ public void saveCriteria(String criteriaName, Criteria criteria, String criteria
...
      */
-    public void findCriteria(Date date1, Date date2) {
-        String firstDate = date1.toString();
-        String secondDate = date2.toString();
-        List list = query.filterByDates(firstDate, secondDate);
+    public List<CriteriaRecord> findCriteria(Date date1, Date date2) {
+        String firstDate = DateSetter.setStartOfDay(date1).toString();
+        String secondDate = DateSetter.setEndOfDay(date2).toString();
+        List filterResult = query.filterByDates(firstDate, secondDate);
+        System.out.println("Filter result " + filterResult.size());
```
Comment (aburgosu) [inline src/main/java/com/fundation/logic/model/QueryForCriteria.java]: Delete print.

File: src/main/java/com/fundation/logic/database/Query.java (hunk 1 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/database/Query.java]: done

File: src/main/java/com/fundation/logic/database/Query.java (hunk 2 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/database/Query.java]: done

File: src/main/java/com/fundation/logic/database/Query.java (hunk 6 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/database/Query.java]: outdated

File: src/main/java/com/fundation/logic/database/Query.java (hunk 7 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/database/Query.java]: done

File: src/main/java/com/fundation/logic/model/QueryForCriteria.java (hunk 9 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/model/QueryForCriteria.java]: done

File: src/main/java/com/fundation/logic/main.java (hunk 8 above)
Comment (MelissaRoman06) [reply src/main/java/com/fundation/logic/main.java]: done

File: src/main/java/com/fundation/logic/database/Query.java [hunk 10]
```diff
@@ -79,30 +82,24 @@ public void deleteByIde(String ID) {
...
-        String lower = secondDate;
-        String upper = firstDate;
-        LocalDateTime lowerdt = LocalDateTime.parse(lower, formatter);
-        LocalDateTime upperdt = LocalDateTime.parse(upper, formatter);
-
-        List<String> infCriterias = new ArrayList<String>();
-        String sql = "SELECT * FROM criterias WHERE date BETWEEN ? AND ?";
+    public List filterByDates(long firstDate, long secondDate) {
+        List<String> infCriteria = new ArrayList<>();
+        String sql = "SELECT * FROM criterias WHERE date >= ? AND date<= ?";
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/database/Query.java]: put space after "date"

File: src/main/java/com/fundation/logic/model/QueryForCriteria.java [hunk 11]
```diff
@@ -47,49 +47,92 @@ public void saveCriteria(String criteriaName, Criteria criteria, String criteria
...
+        long startDate = firstDate.getTime();
+        long endDate = secondDate.getTime();
+        List filterResult = query.filterByDates(startDate, endDate);
+        return registerToCriteriaRecordList(filterResult);
     }
+
+    /**
+     * Allows to get a criteriaRecord by its ID.
+     * @param id
+     * @return
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/model/QueryForCriteria.java]: comment return and param

File: src/main/java/com/fundation/logic/model/QueryForCriteria.java [hunk 12]
```diff
@@ -47,49 +47,92 @@ public void saveCriteria(String criteriaName, Criteria criteria, String criteria
...
+        List<CriteriaRecord> criteriaRecords;
+        criteriaRecords = getAllCriteriaInDB();
+        for (int i = 0; i < criteriaRecords.size(); i++) {
+            if(id.equals(criteriaRecords.get(i).getId())) {
+                return criteriaRecords.get(i);
+            }
+        }
+        return null;
+    }
+
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/model/QueryForCriteria.java]: delete a empty line

File: src/main/java/com/fundation/logic/view/loadSaveCriteria/LoadSavePanel.java [hunk 13]
```diff
@@ -68,11 +77,12 @@ public LoadSavePanel() {
         add(btnDelete, gbc_btnDelete);
     }
 
+
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/view/loadSaveCriteria/LoadSavePanel.java]: delete a empty line

File: src/main/java/com/fundation/logic/database/Query.java [hunk 14]
```diff
@@ -9,34 +9,36 @@
...
     /**
-     * This method inster informations from name and json to Data Base
+     * This method inserts information from name and json to Data Base
      */
     public void insertCriteria(String name, String json) {
         String sql = "INSERT INTO criterias(name, date, json) VALUES(?,?,?)";
         try {
+            Timestamp current = Timestamp.valueOf(LocalDateTime.now());
+            long dateLong = current.getTime();
+            System.out.println(dateLong);
```
Comment (titor-oopart) [inline src/main/java/com/fundation/logic/database/Query.java]: delete this print

**Pull request merged**